test/integration/file_artifact_test.js

```diff
@@ -228,4 +231,67 @@ suite('artifact extration tests', () => {
     // Missing artifact should have an error...
     assert.equal(result.artifacts['public/my-missing.txt'].storageType, 'error');
   });
+
+  test('upload retry', async () => {
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    worker.on('task run', function() {
+      iptables.reject({
```

Comment:
We should clean this up outside of the test in case the test itself fails for some reason we will make sure the iptable rules are cleaned up.